src/main/java/com/fasterxml/jackson/core/json/UTF8JsonGenerator.java

```diff
@@ -1517,14 +1511,17 @@ private final void _writeStringSegment2(final char[] cbuf, int offset, final int
                 outputBuffer[outputPtr++] = (byte) (0xc0 | (ch >> 6));
                 outputBuffer[outputPtr++] = (byte) (0x80 | (ch & 0x3f));
             } else {
-                // multibyte character
-                if (combineSurrogates && isSurrogatePair((char) ch) && offset < end) {
-                    char highSurrogate = (char) ch;
-                    char lowSurrogate = cbuf[offset++];
-                    outputPtr = _outputSurrogatePair(highSurrogate, lowSurrogate, outputPtr);
-                } else {
-                    outputPtr = _outputMultiByteChar(ch, outputPtr);
+                // 3- or 4-byte character
```

Comment:
Only a 4-byte characters (i.e. U+010000 - U+10FFFF). Characters that are encoded as UTF-8 3-byte falls to _outputMultiByteChar method